docs/how-tos/use-multi-format-reader.md

```diff
@@ -0,0 +1,396 @@
+
+`self.hdf5_data` will look like this:
+```python
+{
+    "data/x_values": array([-10.        ,  -9.7979798 ,  -9.5959596 , ...,  10.        ]),
+    "data/y_values": array([3.72665317e-06, 6.14389891e-06, 1.00262383e-05, ..., 3.72665317e-06]),
+    "data/x_units": "eV",
+    "data/y_units": "counts_per_second",
+    "metadata/instrument/version": 1.0,
+    "metadata/instrument/detector/name": "my_gaussian_detector",
+    "metadata/instrument/detector/count_time": 1.2,
+    "metadata/instrument/detector/count_time_units": s",
+}
+```
+## Reading in ELN data
+As we can see in the application definition `NXsimple` above, there are some concepts defined for which there is no equivalent metadata in the HDF5 file. We are therefore using a YAML ELN file to add additional metadata.
+The ELN file `eln_data.yaml` looks like this:
+```yaml
+title: My experiment
+user:
+  name: John Doe
+  address: 123 Science Rd, Data City, DC
+sample:
+  name: my_sample
+  physical_form: powder
+  temperature:
+    value: 300
+    unit: K
+```
+
+It contains metadata about the user and the sample that was measured.
+
+We now need to write a function to read in this ELN data. Luckily, there exists already a solution within `pynxtools`, using the `parse_yaml` function:
+
+```python
+from pynxtools.dataconverter.readers.utils import parse_yml
+
+CONVERT_DICT = {
+    "unit": "@units",
+    "version": "@version",
+    "user": "USER[user]",
+    "instrument": "INSTRUMENT[instrument]",
+    "detector": "DETECTOR[detector]",
+    "sample": "SAMPLE[sample]",
+}
+
+def handle_eln_file(self, file_path: str) -> Dict[str, Any]:
+    self.eln_data = parse_yml(
+        file_path,
+        convert_dict=CONVERT_DICT,
+        parent_key="/ENTRY[entry]",
+    )
+            
+    return {}
+```
+When this method is called, `self.eln_data` will look like this:
+```python
+{
+    "/ENTRY[entry]/title": "My experiment",
+    "/ENTRY[entry]/USER[user]/name": "John Doe",
+    "/ENTRY[entry]/USER[user]/address": "123 Science Rd, Data City, DC",
+    "/ENTRY[entry]/SAMPLE[sample]/name": "my_sample",
+    "/ENTRY[entry]/SAMPLE[sample]/physical_form": "powder",
+    "/ENTRY[entry]/SAMPLE[sample]/temperature": 300,
+    "/ENTRY[entry]/SAMPLE[sample]/temperature/@units": "K"
+}
+```
+Note that here we are using `parent_key="/ENTRY[entry]"` as well as a `CONVERT_DICT`, meaning that each key in `self.eln_data` will start with `"/ENTRY[entry]"` and some of the paths will be converted to match the template notation. This will be important later.
+
+## Parsing the config file
+Next up, we can make us of the config file, which is a JSON file that tells the reader how to map the concepts from the HDF5 and ELN files in order to populate the template designed to match `NXsimple`. Essentially, the config file should contain all keys that are present in the NXDL. In our case, the config file looks like this:

```

Comment:
I recommend to make the statement even stronger, the choices made in the config file define how semantics from the src and trg (NeXus) side are mapped. In other words, the config.json is ontology mapping in practice.

Reply:
added that to the docs describing the design of the multi-format-reader in `learnß 👍